Remove unneeded code and introduce new directory structure

malandrina opened the pull request:
## Why?

We're restructuring our fork of Jarvis to improve its usability and reduce bloat.

## How?

- 9909dc4c17b9578145d5b8785fbdd28a3c889ebb removes GitHub Pages-related code and vendor files
- 46e7db8001263b23b495fb66de69f15007e35982 removes `data.zip`, which doesn't seem to be referenced in the code anywhere
- e1ebca9a19126914d8beb8a53ef0b19deea8d4ae removes pyproject.toml and poetry.lock
- c10b9d2ecbf5c970023c488aaaaf4ab4416afb92 moves code and artifacts related to tool benchmarking and evaluation from `Jarvis` directory to new `evaluation` directory
- a167ff17e4cdd9baf216e492f49b3c41ec0ef9b3 overhauls the directory structure for the call graph generation tool

Before:

```bash
jarviscg main % tree -L 2
.
├── 404.html
├── Gemfile
├── Gemfile.lock
├── Jarvis
│   ├── ReadMe.md
│   ├── code2flow
│   ├── dataset
│   ├── ground_truth
│   ├── reproducing_RQ1
│   ├── reproducing_RQ12_setup
│   ├── reproducing_RQ2
│   └── tool
├── README.md
├── _config.yml
├── _layouts
│   └── default.html
├── assets
│   └── css
├── data.zip
├── evaluation
│   ├── dataset
│   ├── ground_truth
│   ├── reproducing_RQ1
│   └── reproducing_RQ2
├── index.md
├── poetry.lock
├── pyproject.toml
├── src
│   └── jarviscg
└── vendor
    └── bundle

21 directories, 11 files
```

After:

```bash
jarviscg lw-restructure % tree -L 2
.
├── README.md
├── evaluation
│   ├── code2flow
│   ├── dataset
│   ├── ground_truth
│   ├── reproducing_RQ1
│   ├── reproducing_RQ12_setup
│   └── reproducing_RQ2
└── src
    ├── README.md
    ├── jarviscg
    └── pycg
```

## Considerations

The `src` directory is modeled on https://github.com/nuanced-dev/StaticMind. We may decide to make further changes to the organization of the `src` directory as we move forward with creating a jarviscg Python package.

## Testing

I compared the call graph generated for polars/py-polars from the `main` branch to the call graph generated from this branch:

![Screenshot 2025-02-05 at 2 03 21 PM](https://github.com/user-attachments/assets/860e00d9-185d-4a95-9a70-24c49aaa4277)

I have found that Jarvis isn't deterministic in that it doesn't produce exactly the same output every time it runs 🤔 so while that is itself somewhat concerning, the fact that this branch and `main` produced call graphs with 1 line of difference does not bother me.

## Next up

In my next PR I will add a pyproject.toml file defining the jarviscg package and update import statements needed.

spott commented:
I think this is probably good, though it crashed my browser when trying to look at the diff.

It looks like there are a few straggler files that didn't get removed (a .bundle folder and a .DS_Store file in the root of the project), and we should probably add a .gitignore for .DS_Store files.

Otherwise, I'm good to merge.

malandrina commented:
Thanks @spott! ✨ I deleted `.bundle` and updated the .gitignore in my recent commits. Looks like .gitignore includes an entry for `**/.DS_Store` so I left that entry as-is.
